src/lib/storage/constraints/table_constraint_definition.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <vector>
+
+#include "types.hpp"
+
+namespace opossum {
+
+enum class IsPrimaryKey : bool { Yes = true, No = false };
+
+// Defines a constraint on a table. Can optionally be a PRIMARY KEY, requiring the column(s) to be non-NULL.
+// Constraints are currently NOT ENFORCED.
+
+struct TableConstraintDefinition final {
+  TableConstraintDefinition(std::vector<ColumnID> column_ids, const IsPrimaryKey is_primary_key)
+      : columns(std::move(column_ids)), is_primary_key(is_primary_key == IsPrimaryKey::Yes) {
+    DebugAssert(std::is_sorted(columns.begin(), columns.end()), "Expecting Column IDs to be sorted");
+    Assert(std::unique(columns.begin(), columns.end()) == columns.end(), "Expected Column IDs to be unique");
+  }
+
+  std::vector<ColumnID> columns;
+  bool is_primary_key;
```

Comment:
Use `IsPrimaryKey`?

Reply:
Thought about it, but when we access `x.is_primary_key`, we already know what the bool stands for. This is not one of the cases where we will see a bool without its meaning.

Reply:
But adding a new constraint based on a constraint is now `add_constraint(con.column_id, con.is_primary_key ? IsPrimaryKey::Yes : IsPrimaryKey::No)`.